worker/src/Consumer/migrate/drainQueue.ts

```diff
@@ -0,0 +1,41 @@
+import pino from "pino";
+import config from "config";
+import { getInflightMessages, getSchemaVersion, migrateInflightMessagesFromV9 } from "./helpers";
+
+const currentSchema = config.get<string>("Queue.schema");
+
+export async function drainQueue(queue: string, drainSchema: string) {
+  const drainLogger = pino().child({
+    queue,
+    drainSchema,
+  });
+
+  if (currentSchema === drainSchema) {
+    drainLogger.warn(`current ${currentSchema} and drainSchemas ${drainSchema} are the same. Skipping`);
+    return;
+  }
+
+  const schemaVersion = await getSchemaVersion(currentSchema);
+  const drainSchemaVersion = await getSchemaVersion(drainSchema);
+
+  /**
+   * Schema version 20 is equivalent to pg-boss v9.
+   */
+  if (drainSchemaVersion !== 20 || drainSchemaVersion >= schemaVersion) {
+    drainLogger.warn(
+      `drainSchema: ${drainSchema} (v${drainSchemaVersion}) and schema: ${currentSchema} (v${schemaVersion}). Automatic migration is only supported from v20 to v23+`
+    );
+    return;
+  }
+
+  const inflightMessagesInV9 = await getInflightMessages(queue, drainSchema);
+
+  if (!inflightMessagesInV9?.length) {
+    drainLogger.info(`Skipping draining of ${drainSchema}.${queue}. No inflight messages found`);
+    return;
+  }
+
+  drainLogger.info({ messagesToMigrate: inflightMessagesInV9 }, `Migrating ${inflightMessagesInV9.length} to ${currentSchema}`);
```

Comment:
Is there a benefit to printing out the messages that are being migrated? I wonder whether it makes more sense to just make it a count query and just print the number of messages that were found

Reply:
`getInflightMessages` selects the ids only, not the entire message. But the log message does print the number of messages found.
 
it's just a measure for additional cross-checking since `migrateInflightMessagesFromV9` inserts ids from v9